luigi/contrib/bigquery.py

```diff
@@ -325,15 +327,25 @@ def run_job(self, project_id, body, dataset=None):
         new_job = self.client.jobs().insert(projectId=project_id, body=body).execute()
         job_id = new_job['jobReference']['jobId']
         logger.info('Started import job %s:%s', project_id, job_id)
+        poll_counter = 0
         while True:
-            status = self.client.jobs().get(projectId=project_id, jobId=job_id).execute()
-            if status['status']['state'] == 'DONE':
-                if status['status'].get('errors'):
-                    raise Exception('BigQuery job failed: {}'.format(status['status']['errors']))
-                return
-
-            logger.info('Waiting for job %s:%s to complete...', project_id, job_id)
-            time.sleep(5.0)
+            try:
+                status = self.client.jobs().get(projectId=project_id, jobId=job_id).execute()
+                if status['status']['state'] == 'DONE':
+                    if status['status'].get('errors'):
```

Comment:
Errors being present doesn't imply that the job failed, presence of errorResult means that the job failed.
See https://cloud.google.com/bigquery/loading-data#loading_avro_files and example here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/bigquery/cloud-client/load_data_from_file.py#L58